llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3992,6 +3926,32 @@ class VPIRBasicBlock : public VPBasicBlock {
   BasicBlock *getIRBasicBlock() const { return IRBB; }
 };
 
+/// Track information about the canonical IV value of a region.
+class VPCanonicalIVInfo {
+  /// VPRegionValue for the canonical IV. The allocation is managed by
+  /// VPCanonicalIVInfo.
+  std::unique_ptr<VPRegionValue> CanIV;
+  Type *Ty = nullptr;
```

Comment:
It might be good for VPRegionValue to holds its Type, and DL. Probably clearer after additional VPRegionValues are introduced.